main.tf

```diff
@@ -0,0 +1,79 @@
+resource "aws_cloudtrail" "trail" {
+  name           = "trail${replace(title(var.project), " ", "")}${title(var.environment)}"
+  s3_bucket_name = "${var.s3_bucket_name}"
+
+  enable_logging                = true
+  include_global_service_events = true
+  is_multi_region_trail         = false
+  enable_log_file_validation    = false
+
+  tags {
+    Name        = "${var.project}"
+    Environment = "${var.environment}"
+  }
+
+  depends_on = ["aws_s3_bucket.trail"]
+}
+
+resource "aws_s3_bucket" "trail" {
+  count = "${var.create_s3_bucket ? 1 : 0}"
+
+  bucket = "${var.s3_bucket_name}"
+  region = "${var.region}"
+
+  force_destroy = true
+
+  lifecycle_rule {
+    enabled = true
+
+    expiration {
+      days = "${var.s3_bucket_lifecycle_expiration}"
+    }
+  }
+}
+
+resource "aws_s3_bucket_policy" "trail" {
+  count = "${var.create_s3_bucket ? 1 : 0}"
+
+  bucket = "${aws_s3_bucket.trail.id}"
+  policy = "${data.aws_iam_policy_document.cloudtrail_log_access.json}"
+}
+
+#
+# Access policy for CloudTrail <> S3
+# See: https://docs.aws.amazon.com/awscloudtrail/latest/userguide/create-s3-bucket-policy-for-cloudtrail.html
+#
```

Comment:
It would be really nice if this existed as a managed policy, but as far as I can tell it doesn't, possibly because the `resources` sections of each statement need to be pointed to your buckets.